linkcheck/htmlutil/linkparse.py

```diff
@@ -156,10 +156,11 @@ class LinkFinder (TagFinder):
     """Find HTML links, and apply them to the callback function with the
     format (url, lineno, column, name, codebase)."""
 
-    def __init__ (self, callback, tags):
+    def __init__ (self, callback, tags, ignore_classes):
```

Comment:
This is a new required positional parameter.  You've updated one call site, but two more remain:

```
tests/test_linkparser.py:        h = linkparse.LinkFinder(self._test_one_url(url), linkparse.LinkTags)
tests/test_linkparser.py:        h = linkparse.LinkFinder(callback, linkparse.LinkTags)
```

One possibility: make this a keyword argument with the default value of `None`, then you won't have to touch unrelated unit tests.

OTOH by looking at the existing tests for LinkFinder you may discover how to add a new test for this feature.  Or maybe an integration test could be better.